unitTests/ui/OmniboxAnalyticsTest.ts

```diff
@@ -0,0 +1,85 @@
+import { OmniboxAnalytics } from '../../src/ui/Omnibox/OmniboxAnalytics';
+import { IAnalyticsOmniboxSuggestionMeta } from '../../src/ui/Analytics/AnalyticsActionListMeta';
+
+export function OmniboxAnalyticsTest() {
+  let omniboxAnalytics: OmniboxAnalytics;
+
+  describe('omniboxAnalyticsTest', () => {
+    beforeEach(() => {
+      omniboxAnalytics = new OmniboxAnalytics();
+    });
+
+    function random256Array(array: string[]) {
+      while (array.join('').length <= 256) {
+        array.push(
+          Math.random()
+            .toString(36)
+            .substring(2)
+        );
+      }
+    }
+
+    it('return IAnalyticsOmniboxSuggestionMeta when nothing has change since initialisation', () => {
+      expect(omniboxAnalytics.buildCustomDataForPartialQueries()).toEqual(
+        jasmine.objectContaining(<IAnalyticsOmniboxSuggestionMeta>{
+          partialQueries: '',
+          suggestionRanking: undefined,
+          suggestions: '',
+          partialQuery: undefined
+        })
+      );
+    });
+
+    it('suggestionRanking should get set with the right value', () => {
+      omniboxAnalytics.suggestionRanking = 3;
+      expect(omniboxAnalytics.buildCustomDataForPartialQueries()).toEqual(
+        jasmine.objectContaining(<IAnalyticsOmniboxSuggestionMeta>{
+          partialQueries: '',
+          partialQuery: undefined,
+          suggestionRanking: 3,
+          suggestions: ''
+        })
+      );
+    });
+
+    it('partialQueries and suggestions values get separated by a semicolon', () => {
+      omniboxAnalytics.partialQueries = ['foo', 'foo2'];
+      omniboxAnalytics.suggestions = ['bar', 'bar2'];
+      expect(omniboxAnalytics.buildCustomDataForPartialQueries()).toEqual(
+        jasmine.objectContaining(<IAnalyticsOmniboxSuggestionMeta>{
+          partialQueries: 'foo;foo2',
+          partialQuery: 'foo2',
```

Comment:
Why is the partialQuery set to the last element in partialQueries as opposed to the first?

Reply:
so if I write the query `test` and than, need to log an analytics,`partialQueries` will look like: 

`['t', 'te',  'tes', 'test']`

The last thing you write in `partialQueries` is what is in the Omnibox when you log analytics so the successful query is always the last one.